src/network/netdev/wireguard.c

```diff
@@ -0,0 +1,629 @@
+                                       const char *rvalue,
+                                       void *data,
+                                       void *userdata) {
+        union in_addr_union addr;
+        unsigned char prefixlen;
+        int r, family;
+        Wireguard *w;
+        WireguardPeer *peer;
+        WireguardIpmask *ipmask;
+
+        assert(rvalue);
+        assert(data);
+
+        w = WIREGUARD(data);
+
+        peer = wireguard_peer_new(w, section_line);
+        if (!peer)
+                return log_oom();
+
+        for (;;) {
+                _cleanup_free_ char *word = NULL;
+
+                r = extract_first_word(&rvalue, &word, ",", 0);
+                if (r == 0)
+                        break;
+                if (r == -ENOMEM)
+                        return log_oom();
+                if (r < 0) {
+                        log_syntax(unit, LOG_ERR, filename, line, r, "Failed to split allowed ips \"%s\" option: %m", rvalue);
+                        break;
+                }
+
+                r = parse_address_and_prefixlen(word, &family, &addr, &prefixlen);
+                if (r < 0) {
+                        log_syntax(unit, LOG_ERR, filename, line, r, "Network address is invalid, ignoring assignment: %s", word);
+                        return 0;
+                }
+
+                ipmask = new0(WireguardIpmask, 1);
+                if (!ipmask)
+                        return log_oom();
+                ipmask->fields.family = family;
+                ipmask->fields.ip6 = addr.in6;
+                ipmask->fields.cidr = prefixlen;
+
+                LIST_PREPEND(ipmasks, peer->ipmasks, ipmask);
+                w->allocation_size += sizeof(struct wgipmask);
+                peer->fields.num_ipmasks++;
+        }
+
+        return 0;
+}
+
+int config_parse_wireguard_endpoint(const char *unit,
+                                    const char *filename,
+                                    unsigned line,
+                                    const char *section,
+                                    unsigned section_line,
+                                    const char *lvalue,
+                                    int ltype,
+                                    const char *rvalue,
+                                    void *data,
+                                    void *userdata) {
+        Wireguard *w;
+        WireguardPeer *peer;
+        size_t len;
+        const char *begin, *end = NULL;
+        _cleanup_free_ char *host = NULL, *port = NULL;
+        _cleanup_free_ WireguardEndpoint *endpoint = NULL;
+
+        assert(data);
+        assert(rvalue);
+
+        w = WIREGUARD(data);
+
+        assert(w);
+
+        peer = wireguard_peer_new(w, section_line);
+        if (!peer)
+                return log_oom();
+
+        endpoint = new0(WireguardEndpoint, 1);
+        if (!endpoint)
+                return log_oom();
+
+        if (rvalue[0] == '[') {
```

Comment:
I'm pretty sure this parsing algorithm was copy and pasted from a prior git revision of WireGuard. As such, you'll need to retain my copyright line in this file.

Reply:
Yes, please err on the side of too many copyright headers, rather than too few. Even if the code is not similar anymore, but was based on some other code, it's better to mention the original authorship (and add your own (C) line).

Reply:
The authorship was updated in the meantime.

Reply:
As I wrote `set_wireguard_interface` for this commit, my copyright has been added to the top anyway with that, so this comment here is now resolved.